CMakeLists.txt

```diff
@@ -500,6 +500,9 @@ if(EMSCRIPTEN)
   target_link_libraries(binaryen_wasm optimized "-Wno-error=closure")
   target_link_libraries(binaryen_wasm optimized "-flto")
   target_link_libraries(binaryen_wasm debug "--profiling")
+  # Avoid catching exit as that can confuse error reporting in Node,
+  # see https://github.com/emscripten-core/emscripten/issues/17228
+  target_link_libraries(binaryen_wasm "-sNODEJS_CATCH_EXIT=0")
```

Comment:
I'm in the process of trying to disable this by default, but this change lgtm either way.